### ISSUE TITLE
Issues using WickedEngine as submodule

### ISSUE THREAD

**Issue by mlavik1** [turanszkij/WickedEngine#788]
Hi!
First of all, thanks for making this *amazing* project. It has been perfect for my needs so far :)

I'm trying to add WickedEngine as a git submodule, and using `add_subdirectory` in CMake to add it as a subproject that I can link directly to.
This is my project structure:
- MyProject
   - WickedEngine

However, I'm running into some issues due to the use of `CMAKE_SOURCE_DIR` in several CMakeLists.txt files.
Example log from my project:
```
 Error copying file (if different) from "/home/matias/Projects/citygml-viewer/WickedEngine/libdxcompiler.so" to "/home/matias/Projects/citygml-viewer/build/3rdparty/WickedEngine/WickedEngine".
```

`CMAKE_SOURCE_DIR` returns the path to the top level of the source tree, which in my case is the root directory of my project. I believe it might be better to avoid using `CMAKE_SOURCE_DIR` at all, since it adds some limitations.

I was able to solve this issue by setting a new `WICKED_ROOT_DIR` variable in the root CMakeLists.txt and using that instead of `CMAKE_SOURCE_DIR`. Another alternative would be to use relative paths of `CMAKE_CURRENT_SOURCE_DIR`.

I'll make a PR, and then you can decide if you want this change or not :)

By the way: To use use WickedEngine as a subproject I also had to copy the "shaders" directory from "WickedEngine/shaders" into "MyProject/WickedEngine/shaders". I guess making that customizable could also be something worth considering, though it's not that much of an issue.

**Comment by turanszkij** [comment]
Hi, the shaders folder can be overridden if you wat to avoid copying. In your application you can call `wi::renderer::SetShaderSourcePath()` (location where .hlsl shader souce code files are located) and `wi::rederer::SetShaderPath()` (location where compiled shader binary .cso files are located after they are compiled).

I'm not sure about the Cmake stuff as I am personally not using it, but maybe @portaloffreedom can say something about it as it's his creation.

**Comment by mlavik1** [comment]
Thanks for the info! I will try that 😁

And ok, I guess they can have a look at my auggested fix as well then: #789
But no hurry 🙂

Happy holidays!

**Comment by mlavik1** [comment]
Thanks for merging! I'll close this issue then.

**Comment by brakhane** [comment]
@mlavik1 A tip for the future: if you add "fixes #788" or "closes #788" in the commit message, it will automatically close the issue once it gets merge

**Comment by mlavik1** [comment]
> @mlavik1 A tip for the future: if you add "fixes #788" or "closes #788" in the commit message, it will automatically close the issue once it gets merged

Oh, that's convenient! Thanks for the tip :)
